src/framework/standard/mod.rs

```diff
@@ -835,32 +840,39 @@ pub(crate) async fn has_correct_permissions(
 ) -> bool {
     if options.required_permissions().is_empty() {
         true
-    } else if let Some(guild) = message.guild(&cache).await {
-        let channel = match guild.channels.get(&message.channel_id) {
-            Some(channel) => channel,
-            None => return false,
-        };
-        let member = match guild.members.get(&message.author.id) {
-            Some(member) => member,
-            None => return false,
-        };
+    } else {
+        let has_perms_accessor = |guild: &Guild| {
+            let channel_option = guild.channels.get(&message.channel_id);
 
-        let perms = match guild.user_permissions_in(channel, member) {
-            Ok(perms) => perms,
-            Err(e) => {
-                tracing::error!(
-                    "Error getting permissions for user {} in channel {}: {}",
-                    member.user.id,
-                    channel.id,
-                    e
-                );
+            if channel_option.is_none() {
                 return false;
-            },
+            }
+
+            let member_option = guild.members.get(&message.author.id);
+
+            if member_option.is_none() {
+                return false;
+            }
```

Comment:
This can be condensed into one `match` expression just as it was before, removing the need to unwrap the `Option` later with `expect`.
```suggestion
            let member = match guild.members.get(&message.author.id) {
                Some(member) => member,
                None => return false,
            };

```

Reply:
What about `member_option.unwrap_or(false)`?

Reply:
Wow, I'm dumb. I'll change this.